cfgov/legacy/views/housing_counselor.py

```diff
@@ -58,74 +74,13 @@ def get_counselors(cls, request, zipcode):
 
         return response.json()
 
-    @staticmethod
-    def s3_url(format, zipcode):
-        path = settings.HOUSING_COUNSELOR_S3_PATH_TEMPLATE.format(
-            format=format,
-            zipcode=zipcode
-        )
-        return https_s3_url_prefix() + path
-
-    @classmethod
-    def s3_json_url(cls, zipcode):
-        return cls.s3_url(format='json', zipcode=zipcode)
-
-    @classmethod
-    def s3_pdf_url(cls, zipcode):
-        return cls.s3_url(format='pdf', zipcode=zipcode)
-
 
-class HousingCounselorPDFView(View):
+class HousingCounselorPDFView(View, HousingCouncelorS3URLMixin):
     def get(self, request):
         form = HousingCounselorForm(request.GET)
 
         if not form.is_valid():
             return HttpResponseBadRequest('invalid zip code')
 
         zipcode = form.cleaned_data['zip']
-        render_url = self.get_render_url(zipcode)
-        filename = '{}.pdf'.format(zipcode)
-
-        return self.generate_pdf_from_url(render_url, filename)
-
-    def get_render_url(self, zipcode):
-        html_path = reverse('housing-counselor')
-        return 'http://localhost{}?zipcode={}&pdf'.format(html_path, zipcode)
-
-    def generate_pdf_from_url(self, url, filename):
-        result = self.pdfreactor.renderDocumentFromURL(url)
-
-        if result is None:
-            raise Exception('Error while rendering PDF: {}'.format(
-                self.pdfreactor.getError()
-            ))
-
-        response = HttpResponse(result, content_type='application/pdf')
-        response['Content-Disposition'] = \
-            'attachment; filename={0}'.format(filename)
-        return response
-
-    @cached_property
-    def pdfreactor(self):
-        sys.path.append(os.environ['PDFREACTOR_LIB'])
-        from PDFreactor import PDFreactor
-
-        reactor = PDFreactor()
-
-        reactor.setLogLevel(PDFreactor.LOG_LEVEL_FATAL)
-
-        # License key needs to be cast to str here because PDFReactor
-        # doesn't play well with unicode.
-        reactor.setLicenseKey(str(os.environ['PDFREACTOR_LICENSE']))
-
-        reactor.setAuthor('CFPB')
-        reactor.setAddTags(True)
-        reactor.setAddBookmarks(True)
-
-        # Ensure that reactor can be connected to.
-        try:
-            reactor.renderDocumentFromContent('')
-        except Exception:
-            raise RuntimeError('Cannot connect to local PDFReactor')
-
-        return reactor
+        return redirect(self.s3_pdf_url(zipcode))
```

Comment:
@rosskarchner @chosak for illustration, here's a redirect.

Reply:
This looks great - minor typo "councelor".